### DISCUSSION TITLE
Question: Orthographic display and Probing

### DISCUSSION THREAD

**Discussion by jandyman** [winder/Universal-G-Code-Sender#1978] [Help]
I'm looking to upgrade my CNC machine, and in the process switching from UCCNC to GBRL. I've looked at OpenBuildsCONTROL, and a friend told me he uses UGS. So two questions.

Apparently UGS displays toolpaths in perspective mode. Although I can see value in this, for a few different reasons, I find orthographic display to be sort of essential in a CNC controller UI. In order to get orthographic display to work in OpenBuildCONTROL, I will have to dive in and create a somewhat involved macro. What would be involved in getting orthographic display working in UGS?

Second question is about probing. I use an add on to UCCNC that provide for X and Y probing, center probing, and of course simple Z probing, all to establish machine coordinates. Are these facilities either included in UGS or easily added by a competent programmer (which I am)?

I'm a little nervous about the switch from UCCNC to GRBL, with my nervousness centered on the G code sender UI, hence my questions. Any help in advance would be greatly appreciated.

**Comment by breiler** [comment]
> Apparently UGS displays toolpaths in perspective mode. Although I can see value in this, for a few different reasons, I find orthographic display to be sort of essential in a CNC controller UI.

No, UGS uses ortographic projection.

>  all to establish machine coordinates

What do you mean by this? Machine coordinates are established with homing. 
Regarding probing, there is a module for it available in the `Window -> Plugins` menu.

**Reply by jandyman** [reply]
Thanks, good to hear about the orthographic projection. I guess my friend misinformed me. What I meant about probing was maybe using the wrong terminology. If I set up a work coordinate system in (for instance) Fusion 360, I choose an origin. Then I use probing to make that agree with what is on my router table. The simple example is Z probing to deal with bit length, but I also use X and Y and center probing depending on how I've set things up. For example using this add on to UCCNC:

<img width="760" alt="image" src="https://user-images.githubusercontent.com/7635406/189495086-6a295859-b899-4a69-941e-52b202fc38c6.png">

Edit: I just downloaded and am trying UGS, it looks like the thing that is missing is center probing using a conductive ring, which I use a lot. But if writing a plug-in isn't too hard, I could implement it myself.
